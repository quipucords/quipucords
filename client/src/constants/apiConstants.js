const API_RESPONSE_CREDENTIALS_COUNT = 'count';
const API_RESPONSE_CREDENTIALS_RESULTS = 'results';

const API_RESPONSE_CREDENTIAL_CRED_TYPE = 'cred_type';
const API_RESPONSE_CREDENTIAL_NAME = 'name';
const API_RESPONSE_CREDENTIAL_ID = 'id';

const API_RESPONSE_SCANS_COUNT = 'count';
const API_RESPONSE_SCANS_RESULTS = 'results';

const API_RESPONSE_JOBS_RESULTS = 'results';

const API_RESPONSE_SOURCES_COUNT = 'count';
const API_RESPONSE_SOURCES_RESULTS = 'results';

const API_RESPONSE_SOURCE_CREDENTIALS = 'credentials';
const API_RESPONSE_SOURCE_CREDENTIALS_ID = 'id';
const API_RESPONSE_SOURCE_HOSTS = 'hosts';
const API_RESPONSE_SOURCE_ID = 'id';
const API_RESPONSE_SOURCE_NAME = 'name';
const API_RESPONSE_SOURCE_OPTIONS = 'options';
const API_RESPONSE_SOURCE_OPTIONS_SSL_CERT = 'ssl_cert_verify';
const API_RESPONSE_SOURCE_OPTIONS_SSL_PROTOCOL = 'ssl_protocol';
const API_RESPONSE_SOURCE_OPTIONS_DISABLE_SSL = 'disable_ssl';
const API_RESPONSE_SOURCE_OPTIONS_PARAMIKO = 'use_paramiko';
const API_RESPONSE_SOURCE_PORT = 'port';
const API_RESPONSE_SOURCE_SOURCE_TYPE = 'source_type';

const API_SUBMIT_SOURCE_CREDENTIALS = 'credentials';
const API_SUBMIT_SOURCE_HOSTS = 'hosts';
const API_SUBMIT_SOURCE_ID = 'id';
const API_SUBMIT_SOURCE_NAME = 'name';
const API_SUBMIT_SOURCE_OPTIONS = 'options';
const API_SUBMIT_SOURCE_OPTIONS_SSL_CERT = 'ssl_cert_verify';
const API_SUBMIT_SOURCE_OPTIONS_SSL_PROTOCOL = 'ssl_protocol';
const API_SUBMIT_SOURCE_OPTIONS_DISABLE_SSL = 'disable_ssl';
const API_SUBMIT_SOURCE_OPTIONS_PARAMIKO = 'use_paramiko';
const API_SUBMIT_SOURCE_PORT = 'port';
const API_SUBMIT_SOURCE_SOURCE_TYPE = 'source_type';

const apiTypes = {
  API_RESPONSE_CREDENTIALS_COUNT,
  API_RESPONSE_CREDENTIALS_RESULTS,
  API_RESPONSE_CREDENTIAL_CRED_TYPE,
  API_RESPONSE_CREDENTIAL_NAME,
  API_RESPONSE_CREDENTIAL_ID,
  API_RESPONSE_SCANS_COUNT,
  API_RESPONSE_SCANS_RESULTS,
  API_RESPONSE_JOBS_RESULTS,
  API_RESPONSE_SOURCES_COUNT,
  API_RESPONSE_SOURCES_RESULTS,
  API_RESPONSE_SOURCE_CREDENTIALS,
  API_RESPONSE_SOURCE_CREDENTIALS_ID,
  API_RESPONSE_SOURCE_HOSTS,
  API_RESPONSE_SOURCE_ID,
  API_RESPONSE_SOURCE_NAME,
  API_RESPONSE_SOURCE_OPTIONS,
  API_RESPONSE_SOURCE_OPTIONS_SSL_CERT,
  API_RESPONSE_SOURCE_OPTIONS_SSL_PROTOCOL,
  API_RESPONSE_SOURCE_OPTIONS_DISABLE_SSL,
  API_RESPONSE_SOURCE_OPTIONS_PARAMIKO,
  API_RESPONSE_SOURCE_PORT,
  API_RESPONSE_SOURCE_SOURCE_TYPE,
  API_SUBMIT_SOURCE_CREDENTIALS,
  API_SUBMIT_SOURCE_HOSTS,
  API_SUBMIT_SOURCE_ID,
  API_SUBMIT_SOURCE_NAME,
  API_SUBMIT_SOURCE_OPTIONS,
  API_SUBMIT_SOURCE_OPTIONS_SSL_CERT,
  API_SUBMIT_SOURCE_OPTIONS_SSL_PROTOCOL,
  API_SUBMIT_SOURCE_OPTIONS_DISABLE_SSL,
  API_SUBMIT_SOURCE_OPTIONS_PARAMIKO,
  API_SUBMIT_SOURCE_PORT,
  API_SUBMIT_SOURCE_SOURCE_TYPE
};

export {
  apiTypes as default,
  apiTypes,
  API_RESPONSE_CREDENTIALS_COUNT,
  API_RESPONSE_CREDENTIALS_RESULTS,
  API_RESPONSE_CREDENTIAL_CRED_TYPE,
  API_RESPONSE_CREDENTIAL_NAME,
  API_RESPONSE_CREDENTIAL_ID,
  API_RESPONSE_SCANS_COUNT,
  API_RESPONSE_SCANS_RESULTS,
  API_RESPONSE_JOBS_RESULTS,
  API_RESPONSE_SOURCES_COUNT,
  API_RESPONSE_SOURCES_RESULTS,
  API_RESPONSE_SOURCE_CREDENTIALS,
  API_RESPONSE_SOURCE_CREDENTIALS_ID,
  API_RESPONSE_SOURCE_HOSTS,
  API_RESPONSE_SOURCE_ID,
  API_RESPONSE_SOURCE_NAME,
  API_RESPONSE_SOURCE_OPTIONS,
  API_RESPONSE_SOURCE_OPTIONS_SSL_CERT,
  API_RESPONSE_SOURCE_OPTIONS_SSL_PROTOCOL,
  API_RESPONSE_SOURCE_OPTIONS_DISABLE_SSL,
  API_RESPONSE_SOURCE_OPTIONS_PARAMIKO,
  API_RESPONSE_SOURCE_PORT,
  API_RESPONSE_SOURCE_SOURCE_TYPE,
  API_SUBMIT_SOURCE_CREDENTIALS,
  API_SUBMIT_SOURCE_HOSTS,
  API_SUBMIT_SOURCE_ID,
  API_SUBMIT_SOURCE_NAME,
  API_SUBMIT_SOURCE_OPTIONS,
  API_SUBMIT_SOURCE_OPTIONS_SSL_CERT,
  API_SUBMIT_SOURCE_OPTIONS_SSL_PROTOCOL,
  API_SUBMIT_SOURCE_OPTIONS_DISABLE_SSL,
  API_SUBMIT_SOURCE_OPTIONS_PARAMIKO,
  API_SUBMIT_SOURCE_PORT,
  API_SUBMIT_SOURCE_SOURCE_TYPE
};
