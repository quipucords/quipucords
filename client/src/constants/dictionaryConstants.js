const authDictionary = {
  sshKey: 'SSH Key',
  usernamePassword: 'Username and Password'
};

const scanStatusDictionary = {
  success: 'Successful',
  completed: 'Completed',
  failed: 'Failed',
  created: 'Created',
  running: 'Running',
  paused: 'Paused',
  pending: 'Pending',
  canceled: 'Canceled'
};

const srcDictionary = {
  network: 'Network',
  satellite: 'Satellite',
  vcenter: 'VCenter'
};

const dictionary = {
  ...authDictionary,
  ...scanStatusDictionary,
  ...srcDictionary
};

export { dictionary as default, dictionary, authDictionary, scanStatusDictionary, srcDictionary };
