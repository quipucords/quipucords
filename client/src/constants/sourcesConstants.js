
const LOAD_SOURCES_SUCCESS = 'LOAD_SOURCES_SUCCESS';

export {
  LOAD_SOURCES_SUCCESS
}
