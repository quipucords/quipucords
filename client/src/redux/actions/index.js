import * as credentialsActions from './credentialsActions';
import * as factsActions from './factsActions';
import * as reportsActions from './reportsActions';
import * as scansActions from './scansActions';
import * as sourcesActions from './sourcesActions';
import * as statusActions from './statusActions';
import * as userActions from './userActions';

const actions = {
  credentials: credentialsActions,
  facts: factsActions,
  reports: reportsActions,
  scans: scansActions,
  sources: sourcesActions,
  status: statusActions,
  user: userActions
};

const reduxActions = { ...actions };

export {
  reduxActions as default,
  reduxActions,
  actions,
  credentialsActions,
  factsActions,
  reportsActions,
  scansActions,
  sourcesActions,
  statusActions,
  userActions
};
