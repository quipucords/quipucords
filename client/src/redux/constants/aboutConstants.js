const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
