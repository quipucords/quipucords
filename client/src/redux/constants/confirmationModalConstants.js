const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
