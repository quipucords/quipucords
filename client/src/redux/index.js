import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import store from './store';
import reduxActions from './actions';
import reduxReducers from './reducers';
import reduxSelectors from './selectors';
import reduxTypes from './constants';

const connectRouter = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(
    connect(
      mapStateToProps,
      mapDispatchToProps
    )(component)
  );

export { connect, connectRouter, reduxActions, reduxReducers, reduxSelectors, reduxTypes, store };
