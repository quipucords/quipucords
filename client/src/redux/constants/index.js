import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
