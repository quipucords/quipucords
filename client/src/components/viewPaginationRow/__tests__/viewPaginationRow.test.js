import React from 'react';
import { mount } from 'enzyme';
import ViewPaginationRow from '../viewPaginationRow';
import { viewTypes } from '../../../redux/constants';

describe('ViewPaginationRow Component', () => {
  it('should render', () => {
    const props = {
      viewType: viewTypes.SCANS_VIEW,
      currentPage: 1,
      pageSize: 10,
      totalCount: 200,
      totalPages: 20
    };

    const component = mount(<ViewPaginationRow {...props} />);

    expect(component.render()).toMatchSnapshot();
  });
});
