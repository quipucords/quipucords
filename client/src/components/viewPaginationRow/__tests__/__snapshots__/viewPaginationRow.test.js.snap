// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPaginationRow Component should render 1`] = `
<form
  class="content-view-pf-pagination list-view-pagination-top list-view-pf-pagination clearfix"
>
  <div
    class="form-group"
  >
    <div
      class="dropdown pagination-pf-pagesize btn-group"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="dropdown-toggle btn btn-default"
        id="pagination-row-dropdown"
        role="button"
        type="button"
      >
        10 
        <span
          class="caret"
        />
      </button>
      <ul
        aria-labelledby="pagination-row-dropdown"
        class="dropdown-menu"
        role="menu"
      >
        <li
          class="active"
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            10
          </a>
        </li>
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            15
          </a>
        </li>
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            25
          </a>
        </li>
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            50
          </a>
        </li>
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            100
          </a>
        </li>
      </ul>
    </div>
    <span>
      per page
    </span>
  </div>
  <div
    class="form-group"
  >
    <span>
      <span
        class="pagination-pf-items-current"
      >
        1-10
      </span>
       of 
      <span
        class="pagination-pf-items-total"
      >
        200
      </span>
    </span>
    <ul
      class="pagination pagination-pf-back"
    >
      <li
        class="disabled"
      >
        <a
          href="#"
          title="First Page"
        >
          <span
            aria-hidden="true"
            class="fa fa-angle-double-left i"
          />
        </a>
      </li>
      <li
        class="disabled"
      >
        <a
          href="#"
          title="Previous Page"
        >
          <span
            aria-hidden="true"
            class="fa fa-angle-left i"
          />
        </a>
      </li>
    </ul>
    <label
      class="sr-only control-label"
    >
      Current Page
    </label>
    <input
      class="pagination-pf-page form-control"
      type="text"
      value="1"
    />
    <span>
       of 
      <span
        class="pagination-pf-pages"
      >
        20
      </span>
    </span>
    <ul
      class="pagination pagination-pf-forward"
    >
      <li
        class=""
      >
        <a
          href="#"
          title="Next Page"
        >
          <span
            aria-hidden="true"
            class="fa fa-angle-right i"
          />
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="#"
          title="Last Page"
        >
          <span
            aria-hidden="true"
            class="fa fa-angle-double-right i"
          />
        </a>
      </li>
    </ul>
  </div>
</form>
`;
